src/Tec/Plugin.php

```diff
@@ -139,61 +130,132 @@ protected function register_plugin_dependencies() {
 
-	/**
-	 * Get Settings instance.
-	 *
-	 * @return Settings
-	 *
-	 * TODO: Remove if not using settings
-	 */
-	private function get_settings() {
-		if ( empty( $this->settings ) ) {
-			$this->settings = new Settings( $this->get_options_prefix() );
+		// Sanitization
+		$vars = $this->getVars();
+
+		// Bail if not custom iCal export
+		if (
+			! tribe_context()->get( 'ical' )
+			&& $vars['custom'] != 1
+		) {
+			return $repository_args;
+		}
+
+		// Resetting the arguments
+		$repository_args                   = [];
+		$repository_args['order']          = 'ASC';
+		$repository_args['posts_per_page'] = $vars['limit'];
+		$repository_args['paged']          = 1;
+
+		// Check if there is a start_date set
+		if (
+			isset( $vars['start_date'] )
+			&& Date::is_valid_date( $vars['start_date'] )
+		) {
+			$start_date = Date::Build_date_object( $vars['start_date'] );
+		} // If not, fall back to this year's beginning
+		else {
+			$start_date = Date::Build_date_object( date( 'Y' ) . '-01-01' );
+		}
+
+		$start = $start_date->format( 'Y-m-d' );
+
+		// Check if there is an end_date set
+		if (
+			isset( $vars['end_date'] )
+			&& Date::is_valid_date( $vars['end_date'] )
+		) {
+			$end_date = Date::Build_date_object( $vars['end_date'] );
+		} // If there is no end date but there was a start year defined, then till the end of that year
+		elseif (
+			isset( $vars['start_date'] )
+			&& ! empty( $vars['start_date'] )
+		) {
+			$end_date = Date::Build_date_object( $start_date->format( 'Y' ) . '-12-31' );
+		} // If no end date defined, fall back to this year's end
+		else {
+			$end_date = Date::Build_date_object( 'end of the year' );
+		}
+
+		$end = $end_date->format( 'Y-m-d' );
+
+		// If year is defined then use only that.
+		if (
+			isset ( $vars['year'] )
+			&& $vars['year'] >= 1900
+			&& $vars['year'] <= 2100
+		) {
+			$start_date = Date::Build_date_object( $vars['year'] . '-01-01' );
+			$start      = $start_date->format( 'Y-m-d' );
+
+			$end_date = Date::Build_date_object( $vars['year'] . '-12-31' );
+			$end      = $end_date->format( 'Y-m-d' );
 		}
 
-		return $this->settings;
+		$repository_args['date_overlaps'] = [
+			$start,
+			$end,
+		];
+
+		return $repository_args;
 	}
 
 	/**
-	 * Get all of this extension's options.
+	 * Sets the event limit in the feed based on the parameter.
 	 *
-	 * @return array
+	 * @param $count
 	 *
-	 * TODO: Remove if not using settings
+	 * @return int
 	 */
-	public function get_all_options() {
-		$settings = $this->get_settings();
+	function set_limit( $count ) {
+		// Sanitization
+		$vars = $this->getVars();
+
+		if (
+			! tribe_context()->get( 'ical' )
+			|| $vars['custom'] != 1
+		) {
+			return $count;
+		}
+
+		// If limit is -1 or not set, then "unlimited"
+		if (
+			! isset( $vars['limit'] )
+			|| $vars['limit'] == - 1
+		) {
+			return 99999;
+		}
 
-		return $settings->get_all_options();
+		return (int) $vars['limit'];
```

Comment:
```suggestion
		return isset( $vars['limit'] ) ? (int) $vars['limit'] : 20;
```

Reply:
@borkweb I'd be happy to hear some on this as well. :)
Is it like:
If `$vars['limit']` is an integer, then return that, 
otherwise return 20
?

Reply:
This is an example of using "ternary operators". See https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.ternary

The conditional checks that the 'limit' key is indeed set on $vars. If so, use that value. If not, use 20 instead.